api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceProducer.java

```diff
@@ -70,14 +73,28 @@ void init() {
     @Produces
     public MetricsService getMetricsService() {
         if (metricsService == null) {
-            HawkularMetrics.Builder metricsServiceBuilder = new HawkularMetrics.Builder();
+            metricsService = new MetricsServiceCassandra();
             Map<String, String> options = new HashMap<>();
             options.put("cqlport", cqlPort);
             options.put("nodes", nodes);
             options.put("keyspace", keyspace);
 
-            metricsServiceBuilder.withOptions(options);
-            metricsService = metricsServiceBuilder.build();
+            CassandraSession.Builder cassandraSessionBuilder = new CassandraSession.Builder();
+            cassandraSessionBuilder.withOptions(options);
+
+            cassandraSessionBuilder.withInitializationCallback(new FutureCallback<Session>() {
+                @Override
+                public void onSuccess(Session session) {
+                    metricsService.startUp(session);
+                }
+
+                @Override
+                public void onFailure(Throwable t) {
+                    throw new RuntimeException("Error trying to get the Cassandra Session", t);
```

Comment:
You do not want to throw any exceptions in the callback methods. They will get swallowed and the thread running the callback may die silently. If I understand the initialization code, we should not reach the onFailure callback in normal circumstances because the async task blocks indefinitely until it establishes a connection. Maybe we should log an ERROR message here and either set some flag to indicate that server start up failed or shut down the server if possible.